### PR TITLE
clang-tidy fixes pt.7

### PR DESCRIPTION
**Pull request by rdspring1**

### CODE REVIEW

File: csrc/scheduler/registry.cpp [hunk 1]
```diff
@@ -242,7 +242,7 @@ class SchedulerTopologyChecker {
                       // If mapped, and producer is a producer of a reduction,
                       // we can resolve this id
                       ids_to_resolve.erase(
-                          ids_to_resolve.begin() + (entry_i - 1));
+                          ids_to_resolve.begin() + (int64_t)entry_i - 1l);
```
Comment (naoyam) [inline csrc/scheduler/registry.cpp:245]: This looks ugly to me. Does this warning disappear if the type of `entry_i` is changed to `int`?

File: csrc/scheduler/registry.cpp [hunk 2]
```diff
@@ -1953,12 +1953,6 @@ class PersistentKernelScheduler : public SchedulerEntry {
               properties.total_iteration_numel,
               vectorization_factor,
               persistent_buffer_size);
-
```
Comment (naoyam) [inline csrc/scheduler/registry.cpp:1956]: Why is this removed?

File: csrc/scheduler/registry.cpp [hunk 3]
```diff
@@ -1981,7 +1975,7 @@ class PersistentKernelScheduler : public SchedulerEntry {
       return false;
     }
 
-    const int64_t max_used_sms = is_cross_grid
+    const int64_t max_used_sms = is_cross_grid && cross_grid_params.has_value()
```
Comment (naoyam) [inline csrc/scheduler/registry.cpp:1978]: Why is this changed?

File: csrc/scheduler/registry.cpp [hunk 4]
```diff
@@ -2039,7 +2033,7 @@ class PersistentKernelScheduler : public SchedulerEntry {
     // disable the schedule if not evenly divisible on Titan RTX and
     // V100, i.e., compute architecture version 7.
     // TODO: Revisit
-    if (is_cross_grid &&
+    if (is_cross_grid && cross_grid_params.has_value() &&
```
Comment (naoyam) [inline csrc/scheduler/registry.cpp:2036]: same here

File: csrc/scheduler/registry.cpp (hunk 1 above)
Comment (rdspring1) [reply csrc/scheduler/registry.cpp:245]: c++ iterators expect `int64_t`.

```
Error (CLANGTIDY) [bugprone-narrowing-conversions,cppcoreguidelines-narrowing-conversions,-warnings-as-errors] narrowing conversion from 'size_t' (aka 'unsigned long') to signed type '__gnu_cxx::__normal_iterator<std::pair<nvfuser::IterDomain *, nvfuser::IterDomain *> *, std::vector<std::pair<nvfuser::IterDomain *, nvfuser::IterDomain *>>>::difference_type' (aka 'long') is implementation-defined
```

File: csrc/scheduler/registry.cpp (hunk 4 above)
Comment (rdspring1) [reply csrc/scheduler/registry.cpp:2036]: `Error (CLANGTIDY) [bugprone-unchecked-optional-access,-warnings-as-errors] unchecked access to optional value`

File: csrc/scheduler/registry.cpp (hunk 1 above)
Comment (naoyam) [reply csrc/scheduler/registry.cpp:245]: Yes, but does clang-tidy complains when it's `int`?

File: csrc/scheduler/registry.cpp (hunk 2 above)
Comment (rdspring1) [reply csrc/scheduler/registry.cpp]: I added it back.

File: csrc/scheduler/registry.cpp (hunk 3 above)
Comment (naoyam) [reply csrc/scheduler/registry.cpp:1985]: There's an assertion inside a if block with `is_cross_grid`, so it's guaranteed to have a value when `is_cross_grid`. If that's not enough for clang-tidy, let's put an assertion again since it doesn't make any sense to have `is_cross_grid` being `true` and `cross_grid_params.has_value()` being `false`.

File: csrc/scheduler/registry.cpp (hunk 1 above)
Comment (rdspring1) [reply csrc/scheduler/registry.cpp]: I thought we were sticking to `int64_t`.

Comment (naoyam) [reply csrc/scheduler/registry.cpp]: Yeah, that's right. Is the `l` suffix necessary?

Comment (naoyam) [reply csrc/scheduler/registry.cpp]: Looks like you have already changed to just cast `entry_i`.

**Comment by naoyam** [comment]
LGTM. Can't "approve" as it's not based off main.

File: csrc/scheduler/normalization.cpp [hunk 5]
```diff
@@ -259,9 +258,9 @@ std::shared_ptr<ReductionParams> innerPersistentHeuristic(
   }
 
   // start from small block size to minimize expensive inter-threads reduction
-  const int threads_after_vectorize =
+  const int64_t threads_after_vectorize =
       inner_most_dimension_numel / inner_reduction_unroll_factor;
-  constexpr int scheduler_per_sm = 4;
+  constexpr int64_t scheduler_per_sm = 4ll;
```
Comment (naoyam) [inline csrc/scheduler/normalization.cpp:263]: Does it need two `l`'s?

File: csrc/scheduler/registry.cpp [hunk 6]
```diff
@@ -1999,6 +1999,8 @@ class PersistentKernelScheduler : public SchedulerEntry {
       }
     }
 
+    TORCH_INTERNAL_ASSERT(is_cross_grid && cross_grid_params.has_value())
```
Comment (naoyam) [inline csrc/scheduler/registry.cpp:2002]: I don't think this is correct. This should be:

```
if (is_cross_grid) {
   TORCH_INTERNAL_ASSERT(cross_grid_params.has_value());
}
```

File: csrc/scheduler/normalization.cpp [hunk 7]
```diff
@@ -389,9 +388,9 @@ std::shared_ptr<ReductionParams> innerPersistentHeuristic(
   // block by a factor of 2
   if (batches_per_block_outer_reduction * batches_per_block_inner_reduction *
               inner_reduction_unroll_factor * outer_reduction_unroll_factor *
-              4 >
-          255 * 3 &&
-      bdimx * bdimy * bdimz * 2 <= max_threads_in_block &&
+              4ll >
+          255ll * 3ll &&
+      bdimx * bdimy * bdimz * 2ll <= max_threads_in_block &&
```
[the remark this answers is not in the document]
Comment (rdspring1) [reply csrc/scheduler/normalization.cpp:393]: When I removed `255ll *3ll`, I see these errors:
> Lint for csrc/scheduler/normalization.cpp:

 Error (CLANGTIDY) [bugprone-implicit-widening-of-multiplication-result,-warnings-as-errors] performing an implicit widening conversion to type 'long long' of a multiplication performed in type 'int'

         389  |  if (batches_per_block_outer_reduction * batches_per_block_inner_reduction *
         390  |              inner_reduction_unroll_factor * outer_reduction_unroll_factor *
         391  |              4ll >
    >>>  392  |          255 * 3 &&
         393  |      bdimx * bdimy * bdimz * 2ll <= max_threads_in_block &&
         394  |      batches_per_block_inner_reduction >
         395  |          batches_per_block_inner_reduction_max) {

Error (CLANGTIDY) [bugprone-narrowing-conversions,cppcoreguidelines-narrowing-conversions,-warnings-as-errors] narrowing conversion from 'int64_t' (aka 'long') to signed type 'int' is implementation-defined

         490  |
         491  |  auto rparams = std::make_shared<ReductionParams>();
         492  |
    >>>  493  |  rparams->cparams.maxrregcount = nvrtc_register_per_thread;
         494  |  rparams->persistent_kernel = true;
         495  |  rparams->fastest_dim = true;
         496  |

[the remark this answers is not in the document]
Comment (rdspring1) [reply csrc/scheduler/normalization.cpp:393]: The double `l` isn't needed here.

[the remark this answers is not in the document]
Comment (naoyam) [reply csrc/scheduler/normalization.cpp:393]: Yes, I understand why clang-tidy complains and why `255ll * 3ll` is a workaround. It just feels bizarre that we can't just write `255 * 3` since it's just a simple integer math expression. Maybe this is a price to pay to use clang-tidy as it certainly has real benefits.

File: csrc/scheduler/registry.cpp (hunk 6 above)
Comment (rdspring1) [reply csrc/scheduler/registry.cpp:2002]: How about this? 
`TORCH_INTERNAL_ASSERT(!is_cross_grid || cross_grid_params.has_value())`

File: csrc/scheduler/normalization.cpp (hunk 7 above)
[the remark this answers is not in the document]
Comment (rdspring1) [reply csrc/scheduler/normalization.cpp:393]: It seems annoying that the default integer types in the cpp standard are `size_t` and `int64_t`, but default type for constants is `int`.

File: csrc/scheduler/registry.cpp (hunk 6 above)
Comment (naoyam) [reply csrc/scheduler/registry.cpp:2002]: Yes, that should be fine too.

**Comment by rdspring1** [comment]
!build

**Comment by rdspring1** [comment]
!build

**Comment by rdspring1** [comment]
!build

**Review by naoyam — APPROVED** [review]
LGTM. Thanks for the fix

**Comment by rdspring1** [comment]
!build
